### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#2]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 One approach we can take is to switch to using a custom-defined callback that utilizes your Logger to get the experiment name and include it in the saved model name. A simple implementation would look like the following:

```python
from pytorch_lightning.callbacks import ModelCheckpoint
from pytorch_lightning import Callback

class CustomModelCheckpoint(Callback):

    def __init__(self, monitor='val_loss', verbose=False, save_top_k=1, mode='auto'):
        self.monitor = monitor
        self.verbose = verbose
        self.save_top_k = save_top_k
        self.mode = mode

    def on_validation_end(self, trainer, pl_module):
        epoch = trainer.current_epoch
        logs = trainer.callback_metrics
        filename = f"{trainer.logger.experiment.name}-epoch{epoch}-{logs[self.monitor]:.2f}"
        trainer.checkpoint_callback = ModelCheckpoint(filepath=filename, 
                                                      save_top_k=self.save_top_k, 
                                                      verbose=self.verbose, 
                                                      monitor=self.monitor, 
                                                      mode=self.mode)
```
You can then replace the original `ModelCheckpoint` callback with the custom one in your `Trainer` initialisation:

```python
trainer = Trainer(callbacks=[CustomModelCheckpoint()])
```

This should allow you to incorporate your WandB experiment name into the saved model's filename.

Note: This solution assumes that your WandB logger is defined and you have the necessary permissions to access `trainer.logger.experiment.name`. Please ensure these preconditions for the smooth functioning of this solution. 

Remember to replace your logger name and WandB logger with your actual instantiated loggers. Additionally, make sure your validation step is correctly logging the metrics you want to monitor.

Finally, note that this uses the `trainer.callback_metrics` dictionary, which stores the outputs of the validation step. The keys in this dictionary should match your logging keys, e.g., `"val_loss"`. 

And it's advised to verify this code as it's not thoroughly tested. Let me know if the problem persists.
